apps/myaccount/src/components/consents/consent-edit.tsx

```diff
@@ -50,38 +58,194 @@ export const AppConsentEdit: FunctionComponent<EditConsentProps> = (
+        deniedPIIClaimList,
+        onPIIClaimToggle,
         ["data-testid"]: testId
     } = props;
     const { t } = useTranslation();
 
     /**
-     * Checks if the PII category is revoked.
+     * A predicate that checks if the PII category is denied.
+     * It iterates through the {@link deniedPIIClaimList} state hook to check
+     * whether there's any mapping category is present.
      *
-     * @param {number} claimId - claim id ie. piiCategoryId.
+     * Important Note: -
+     * We don't use a separate predicate function to check non-revoked
+     * or accepted PII categories. The functionality can be achieved via this
+     * function itself because if a {@code PIICategoryClaimToggleItem} is
+     * not available in {@link deniedPIIClaimList} it is guaranteed that
+     * it is available in {@link acceptedPIIClaimList} vice versa.
+     *
+     * @param {number} piiCategoryId - PII Claim ID {@code purposes.each(purpose.piiCategory.each(pii.id))}
+     * @param {number} purposeId - Purpose ID {@code services.each(purposes.each(purpose.purposeId))}
+     * @param {number} consentReceiptID - This Component's {@code editingConsent} Model
      */
-    const isRevoked = (claimId: number): boolean => {
-        for (const item of revokedClaimList) {
-            if (item.id === editingConsent.consentReceiptID) {
-                return !!item.revoked.includes(claimId);
-            }
+    const isRevoked = (piiCategoryId: number, purposeId: number, consentReceiptID: string): boolean => {
+        for (const deniedPIIItem of deniedPIIClaimList) {
+            if (piiCategoryId === deniedPIIItem.piiCategoryId &&
+                purposeId === deniedPIIItem.purposeId &&
+                consentReceiptID === deniedPIIItem.receiptId) return true;
         }
-    };
+        return false;
+    }
 
     /**
      * Checks if the consent is updatable.
      *
      * @return {boolean}
      */
     const isUpdatable = (): boolean => {
-        for (const item of revokedClaimList) {
-            if (item.id === editingConsent.consentReceiptID) {
-                return item.revoked && item.revoked.length && item.revoked.length > 0;
+
+        // This consent editing view's model {@link editingConsent}
+        const recordOnModelReceipt = chain(editingConsent.consentReceipt?.services || [])
+            .map((service: ServiceInterface) => service.purposes)
+            .flatten()
+            .map((purpose: PurposeInterface) => ( {
+                piiCategory: purpose.piiCategory as PIICategoryWithStatus[],
+                purposeId: purpose.purposeId
+            } ))
+            .flatten()
+            .value();
+
+        // Filter out the piiClaims of this receipt.
+        const recordOnUserInterface = chain([ ...deniedPIIClaimList, ...acceptedPIIClaimList ])
+            .filter((piiClaim) => piiClaim.receiptId === editingConsent.consentReceiptID)
+            .value();
+
+        // FIXME: solve in linear time
```

Comment:
Shall we add a `TODO` and create an issue and link here?

Reply:
A new issue has been created. https://github.com/wso2/product-is/issues/10669